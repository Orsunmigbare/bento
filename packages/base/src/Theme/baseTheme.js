import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
