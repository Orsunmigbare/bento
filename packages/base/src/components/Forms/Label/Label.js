/**
 * @class Label
 */

import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { themify, font, spacing, fontFamily } from "@staccx/theme"

export const LABEL = "LABEL"

const Label = ({ className, children, ...rest }) => (
  <LabelElement className={className} {...rest}>
    {children}
  </LabelElement>
)

const LabelElement = styled.label`
  font-weight: bold;
  font-family: ${fontFamily.body};
  font-size: ${font.base};
  line-height: 1.2;
  padding-bottom: ${spacing.tiny};
  display: block;
  ${themify(LABEL)};
`

Label.defaultProps = {
  className: ""
}

Label.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element,
    PropTypes.string,
    PropTypes.array
  ]).isRequired,
  className: PropTypes.string
}

/** @component */
export default Label
