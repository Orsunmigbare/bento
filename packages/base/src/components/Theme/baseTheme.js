export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
