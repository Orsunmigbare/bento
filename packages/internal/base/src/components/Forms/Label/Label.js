/**
 * @class Label
 */

import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import {
  themify,
  font,
  spacing,
  fontFamily,
  fontWeight,
  themeProps
} from "@staccx/theme"
import themePropTypes from "../../../constants/themePropTypes"

const Label = ({ className, children, ...rest }) => (
  <LabelElement className={className} {...rest}>
    {children}
  </LabelElement>
)

Label.themeProps = {
  label: {
    name: "LABEL",
    description: "Label style",
    type: themePropTypes.style
  }
}

const LabelElement = styled.label`
  font-weight: ${fontWeight.bold};
  font-family: ${fontFamily.body};
  font-size: ${font.base};
  line-height: 1.2;
  padding-bottom: ${spacing.tiny};
  display: block;
  ${themify(Label.themeProps.label)};
`

Label.defaultProps = {
  className: ""
}

Label.propTypes = {
  children: themeProps.children.isRequired,
  className: PropTypes.string
}

/** @component */
export default Label
