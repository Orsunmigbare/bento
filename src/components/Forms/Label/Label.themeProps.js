import { themePropTypes } from "../../../theming"

const themeProps = {
  label: {
    name: "LABEL",
    description: "Label style",
    type: themePropTypes.style
  }
}

export default themeProps
