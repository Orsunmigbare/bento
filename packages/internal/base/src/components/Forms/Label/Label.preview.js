import React from "react"
import Label from "./Label"

const preview = {
  title: "Label",
  category: "component/Forms",
  component: Label,
  render: props => <Label {...props}>This is a label</Label>
}

export default preview
