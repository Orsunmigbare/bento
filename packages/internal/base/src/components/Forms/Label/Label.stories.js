import React from "react"
import { storiesOf } from "@storybook/react"
import Label from "./Label"
import docs from "./Label.md"

storiesOf("components|base/Forms/Label", module)
  .add("No props", () => <Label>This is a label</Label>, {
    info: {
      text: docs
    }
  })
  .add("with props", () => <Label className={"a Label"}>This is label</Label>)
