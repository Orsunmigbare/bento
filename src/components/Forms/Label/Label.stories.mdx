import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Label from "./Label"

<Meta title="Components|Forms/Label" component={Label}/>


# Label
Label defines an element, often input fields or loaders and acts as a description for that element

# import

```javascript
import {Label} from "@staccx/bento"
```

# Types of Labels


## Standard Label

<Preview>
  <Story name="Standard">
   <Label >This is a label</Label>
  </Story>
</Preview>



# Props for Label
<Props of={Label} />
