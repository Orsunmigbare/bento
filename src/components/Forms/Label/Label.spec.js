import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import Label from "./Label"
import { render, screen } from "@testing-library/react"

describe("Label", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Label>This is a label</Label>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("No Children", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Label />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})

describe("Renderer", () => {
  it("Should render", () => {
    render(
      <ThemeProvider theme={baseTheme}>
        <Label data-testid="testID" />
      </ThemeProvider>
    )
    expect(screen.getByTestId("testID")).toBeInTheDocument()
  })
})
