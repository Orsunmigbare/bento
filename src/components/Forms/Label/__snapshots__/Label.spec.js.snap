// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Snapshots No Children 1`] = `
<label
  className="sc-AxjAm hjwNwx"
/>
`;

exports[`Label Snapshots No props 1`] = `
<label
  className="sc-AxjAm hjwNwx"
>
  This is a label
</label>
`;
